test/jdk/jdk/jfr/event/io/TestSerializationMisdeclarationEvent.java

```diff
@@ -0,0 +1,259 @@
+                arguments(EnumClass.class, SUID_FINAL),
+                arguments(EnumClass.class, SER_PERS_INEFFECTIVE_ENUM),
+                arguments(EnumClass.class, SER_PERS_PRIVATE),
+                arguments(EnumClass.class, SER_PERS_STATIC),
+                arguments(EnumClass.class, SER_PERS_FINAL),
+                arguments(EnumClass.class, SER_PERS_TYPE_OSF_ARRAY),
+                arguments(EnumClass.class, PRIV_METH_PRIV),
+                arguments(EnumClass.class, PRIV_METH_NON_STATIC),
+                arguments(EnumClass.class, PRIV_METH_RET_TYPE),
+                arguments(EnumClass.class, PRIV_METH_PARAM_TYPES),
+                arguments(EnumClass.class, PRIV_METH_INEFFECTIVE_ENUM),
+                arguments(EnumClass.class, ACC_METH_INEFFECTIVE_ENUM),
+
+                arguments(RecordClass.class, SER_PERS_INEFFECTIVE_RECORD),
+                arguments(RecordClass.class, SER_PERS_TYPE_OSF_ARRAY),
+                arguments(RecordClass.class, SER_PERS_VALUE_OSF_ARRAY),
+                arguments(RecordClass.class, PRIV_METH_INEFFECTIVE_RECORD),
+
+                arguments(C.class, ACC_METH_NON_ACCESSIBLE),
+
+                arguments(Acc.class, ACC_METH_NON_ABSTRACT),
+                arguments(Acc.class, ACC_METH_NON_STATIC),
+                arguments(Acc.class, ACC_METH_RET_TYPE),
+                arguments(Acc.class, ACC_METH_PARAM_TYPES),
+        };
+    }
+
+    static Arguments[] testGoodClass() {
+        return new Arguments[] {
+                arguments(A.class),
+                arguments(B.class),
+        };
+    }
+
+
+    @ParameterizedTest
+    @MethodSource
+    public void testSingleClassMisdeclarations(Class<?> cls, int kind) {
+        singleClassEvent(cls, kind);
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    public void testGoodClass(Class<?> cls) {
+        assertEquals(0, getEventsFor(cls).size());
+    }
+
+    private static void doLookups() {
+        ObjectStreamClass.lookup(NoSUID.class);
+        ObjectStreamClass.lookup(NonLongSUID.class);
+        ObjectStreamClass.lookup(EnumClass.class);
+        ObjectStreamClass.lookup(RecordClass.class);
+        ObjectStreamClass.lookup(Acc.class);
+
+        ObjectStreamClass.lookup(A.class);
+        ObjectStreamClass.lookup(B.class);
+        ObjectStreamClass.lookup(C.class);
+    }
+
+    private static void singleClassEvent(Class<?> cls, int kind) {
+        assertEquals(1, getEventsFor(cls, kind).size());
+    }
+
+    private static List<RecordedEvent> getEventsFor(Class<?> cls, int kind) {
+        return events.stream()
+                .filter(e -> e.getClass("cls").getName().equals(cls.getName())
+                        && e.getInt("kind") == kind)
+                .toList();
+    }
+
+    private static List<RecordedEvent> getEventsFor(Class<?> cls) {
+        return events.stream()
+                .filter(e -> e.getClass("cls").getName().equals(cls.getName()))
+                .toList();
+    }
+
+    static {
```

Comment:
It's a bit odd for a test case to be doing this in a static block. Could this instead be done in a `org.junit.jupiter.api.BeforeAll` method:

```java
import org.junit.jupiter.api.BeforeAll;
...
@BeforeAll
static void recordEvents() {
  .... // that static block's code here
}
```
Any (unexpected) failures in that method will then be reported in a better way by the testing framework instead of an `ExceptionInInitializer` that would be reported from a failure in static block and the test class itself failing to load.

Reply:
Yes, yours is a cleaner way to go. Will do.